nvme-print.c

```diff
@@ -2316,10 +2316,12 @@ static void json_print_nvme_subsystem_list(nvme_root_t r, unsigned int nsid)
 		nvme_subsystem_t s;
 
 		host_attrs = json_create_object();
-		json_object_add_value_string(host_attrs, "HostNQN",
-					     nvme_host_get_hostnqn(h));
-		json_object_add_value_string(host_attrs, "HostID",
-					     nvme_host_get_hostid(h));
+		if (nvme_host_get_hostnqn(h))
+			json_object_add_value_string(host_attrs, "HostNQN",
```

Comment:
Alright, this makes it behave in the same way as 1.x.  But in 1x branch I see that also `Controller`, `Transport`, `Address` and `State` are handled this way. Would you mind updating these as well?

Reply:
Yep, this is a bit of a whac-o-mole situation. Maybe it would be better to teach json_object_add_value_string() to handle NULL pointers.

Reply:
This really is a bug a in the code. With 2.x we _always_ should fill out 'hostnqn' and 'hostid', as this it the primary search key for the 'nvme_host_t' structure, and without the 'hostnqn' we won't be able to find it in the tree.
So I'm not sure if it's worth fixing that in the json code.

Reply:
I agree. If we want to protect against these kind errors, it would be better to make json_object_add_value_string() & friends more robust.